internal/grpc/server/server.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/mikewurtz/taskman/certs"
+	pb "github.com/mikewurtz/taskman/gen/proto"
+	basegrpc "github.com/mikewurtz/taskman/internal/grpc"
+)
+
+// Wraps the grpcServer and listener together
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+}
+
+// NewServer sets up the gRPC server and listener with mTLS authentication using TLS v1.3
+// Includes interceptors for injecting the client CN into the context for unary and stream calls
+func NewServer(serverAddr string) (*Server, error) {
```

Comment:
Clients will refer to this function as `server.NewServer`. WDYT about `server.New` instead?

(Same comment applies to `client.NewClient`)

Reply:
I like it, it avoids confusion with the function names for grpc.NewServer and grpc.NewClient.